facade/src/main/java/org/jboss/pnc/facade/deliverables/DeliverableAnalyzerManagerImpl.java

```diff
@@ -242,22 +249,85 @@ public void clear(int id) {
         milestone.getDeliveredArtifacts().clear();
     }
 
-    private org.jboss.pnc.model.Artifact findOrCreateNotFoundArtifact(Artifact artifact, TargetRepository targetRepo) {
-        return findOrCreateArtifact(mapNotFoundArtifact(artifact), targetRepo);
+    private Optional<org.jboss.pnc.model.Artifact> getBestMatchingArtifact(
+            Collection<org.jboss.pnc.model.Artifact> artifacts) {
+        if (artifacts == null || artifacts.isEmpty()) {
+            return Optional.empty();
+        }
+
+        if (artifacts.size() == 1) {
+            return Optional.of(artifacts.iterator().next());
```

Comment:
If you want it shorter :smile: 
```suggestion
            return artifacts.stream().findFirst();
```